packages/framework-types/src/searcher.ts

```diff
@@ -76,60 +73,17 @@ export class Searcher<TObject> {
    * Do the actual search by sending all the configured filters to the provided search function
    */
   public async search(): Promise<Array<TObject>> {
-    const searchResult = await this.searcherFunction(this.objectClass.name, this.filters)
+    const searchResult = await this.searcherFunction(
+      this.objectClass.name,
+      this.filters,
+      this._limit,
+      this._afterCursor,
+      this._paginatedVersion
+    )
     return searchResult as Array<TObject>
```

Comment:
Would this fail if the search was paginated? The search result would be the `ReadModelListResult`, and not an array.

Reply:
Also, the user should get the cursor back so that they can use it in `Searcher.afterCursor`